server/src/main/java/org/tctalent/server/service/db/impl/CandidateServiceImpl.java

```diff
@@ -405,2392 +405,2410 @@
+      } else {
+        //Update twin from candidate
+        twin.copy(candidate, textExtracter);
+      }
+    }
+    twin = candidateEsRepository.save(twin);
+    textSearchId = twin.getId();
+
+    //Update textSearchId on candidate if necessary
+    if (!textSearchId.equals(originalTextSearchId)) {
+      candidate.setTextSearchId(textSearchId);
+      candidate = candidateRepository.save(candidate);
+    }
+    return candidate;
+  }
+
+  @Override
+  @NonNull
+  public String getCandidateSubfolderName(CandidateSubfolderType type) {
+    return candidateSubfolderNames.get(type);
+  }
+
+  @Override
+  @Nullable
+  public String getCandidateSubfolderlink(Candidate candidate, CandidateSubfolderType type) {
+    String link = switch (type) {
+      case address -> candidate.getFolderlinkAddress();
+      case character -> candidate.getFolderlinkCharacter();
+      case employer -> candidate.getFolderlinkEmployer();
+      case engagement -> candidate.getFolderlinkEngagement();
+      case experience -> candidate.getFolderlinkExperience();
+      case family -> candidate.getFolderlinkFamily();
+      case identity -> candidate.getFolderlinkIdentity();
+      case immigration -> candidate.getFolderlinkImmigration();
+      case language -> candidate.getFolderlinkLanguage();
+      case medical -> candidate.getFolderlinkMedical();
+      case qualification -> candidate.getFolderlinkQualification();
+      case registration -> candidate.getFolderlinkRegistration();
+    };
+    return link;
+  }
+
+  @Override
+  public void setCandidateSubfolderlink(Candidate candidate, CandidateSubfolderType type,
+      @Nullable String link) {
+
+    switch (type) {
+      case address -> candidate.setFolderlinkAddress(link);
+      case character -> candidate.setFolderlinkCharacter(link);
+      case employer -> candidate.setFolderlinkEmployer(link);
+      case engagement -> candidate.setFolderlinkEngagement(link);
+      case experience -> candidate.setFolderlinkExperience(link);
+      case family -> candidate.setFolderlinkFamily(link);
+      case identity -> candidate.setFolderlinkIdentity(link);
+      case immigration -> candidate.setFolderlinkImmigration(link);
+      case language -> candidate.setFolderlinkLanguage(link);
+      case medical -> candidate.setFolderlinkMedical(link);
+      case qualification -> candidate.setFolderlinkQualification(link);
+      case registration -> candidate.setFolderlinkRegistration(link);
+    }
+  }
+
+  @Override
+  public Candidate createCandidateFolder(long id)
+      throws NoSuchObjectException, IOException {
+    Candidate candidate = getCandidate(id);
+
+    GoogleFileSystemDrive candidateDrive;
+    GoogleFileSystemFolder folder;
+
+    String folderlink = candidate.getFolderlink();
+
+    //If we already have a folderlink stored, that gives us the folder object
+    if (folderlink != null
+        //Avoid any crap links - eg blank field which is not null
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Usage of unprofessional language detected - crap.